test/Integration/tpp-transpose-bf16.mlir

```diff
@@ -0,0 +1,525 @@
+// RUN: tpp-run --vector-To-XSMM %s -e entry -entry-point-result=void -print --seed 123 2>&1 | FileCheck %s
+// RUN: tpp-run --linalg-to-loops %s -e entry -entry-point-result=void -print --seed 123 2>&1 | FileCheck %s
+
+func.func @entry(%arg0 : tensor<32x32xbf16>, %arg1 : tensor<16x32x2xbf16>)-> tensor<16x32x2xbf16> {
+  %expand_shape = tensor.expand_shape %arg0 [[0, 1], [2]] output_shape[16, 2, 32]
+    : tensor<32x32xbf16>
+    into tensor<16x2x32xbf16>
+  %retval = linalg.transpose ins(%expand_shape : tensor<16x2x32xbf16>)
+    outs(%arg1 : tensor<16x32x2xbf16>) permutation = [0, 2, 1]
+  return %retval: tensor<16x32x2xbf16>
+}
+
+// CHECK:	( 0, 0.186523 )
+// CHECK:	( 0.129883, 0.111816 )
+// CHECK:	( 0.151367, 0.388672 )
+// CHECK:	( 0.0106201, 0.0942383 )
+// CHECK:	( 0.000375748, 0.188477 )
+// CHECK:	( 0.298828, 0 )
+// CHECK:	( 0.0981445, 0.144531 )
+// CHECK:	( 0.0112305, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0.0500488, 0.228516 )
+// CHECK:	( 0.128906, 0 )
+// CHECK:	( 0.0148315, 0.249023 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0.00491333 )
+// CHECK:	( 0, 0.253906 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0.0991211 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0, 0.0256348 )
+// CHECK:	( 0.15625, 0 )
+// CHECK:	( 0, 0.104492 )
+// CHECK:	( 0, 0 )
+// CHECK:	( 0.131836, 0 )
+// CHECK:	( 0.207031, 0 )
+// CHECK:	( 0, 0.0991211 )
+// CHECK:	( 0.0649414, 0.242188 )
+// CHECK:	( 0.154297, 0.171875 )
```

Comment:
you don't need to check against ALL values, just the first dozen or so.

Reply:
I didnt see this comment before, we might end up passing incorrect size arguments to xsmm call which might go unnoticed if we dont check ALL values. Am i missing something?

Reply:
@rengolin what do you think? Am i missing something here?

Reply:
as discussed, I have reduced the problem size so that the results are smaller and can be fully compared.